src/coord/cartesian/Grid.ts

```diff
@@ -180,17 +180,18 @@ class Grid implements CoordinateSystemMaster {
 
         this._rect = gridRect;
 
-        const axesList = this._axesList;
-
-        adjustAxes();
+        // sort axes to make sure we estimate ordinal axes' label dimensions after we adjusted
+        // the available grid space based on the value axes' label dimensions
+        const axesList = [...this._axesList].sort((a, b) => axisOrder(a) - axisOrder(b));
```

Comment:
Part 1 of the "magic coefficient" bug fix referenced [earlier](https://github.com/apache/echarts/pull/19348/files#r1411182163)...